app/models/user.rb

```diff
@@ -1,3 +1,8 @@
 class User < ActiveRecord::Base
+  validates :username, presence: true
+  validates :email_address, presence: true
+  validates :email_address, uniqueness: true
+  validates :display_name, length: { in: 2..32 }, allow_blank: true
+  validates_format_of :email_address, with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/, on: :create
```

Comment:
Line is too long. [101/80]<br>Prefer the new style validations `validates :column, format: value` over `validates_format_of`.